django/db/backends/sqlite3/schema.py

```diff
@@ -168,27 +168,25 @@ def altered_table_name(model, temporary_table_name):
             yield
             model._meta.db_table = original_table_name
 
-        # Rename the old table to something temporary
-        old_table_name = model._meta.db_table + "__old"
-        with altered_table_name(model, old_table_name):
+        with altered_table_name(model, model._meta.db_table + "__old"):
+            # Rename the old table to make way for the new
             self.alter_db_table(model, temp_model._meta.db_table, model._meta.db_table)
 
-        # Create a new table with that format. We remove things from the
-        # deferred SQL that match our table name, too
-        self.deferred_sql = [x for x in self.deferred_sql if temp_model._meta.db_table not in x]
-        self.create_model(temp_model)
+            # Create a new table with the updated schema. We remove things
+            # from the deferred SQL that match our table name, too
+            self.deferred_sql = [x for x in self.deferred_sql if temp_model._meta.db_table not in x]
+            self.create_model(temp_model)
```

Comment:
``self.create_model(temp_model)`` would break tests before #4130 if called inside the context manager, because ``temp_model`` could still hold a reference to ``model`` in its fields.